waspc/data/Generator/templates/react-app/src/auth/pages/OAuthCodeExchange.js

```diff
@@ -0,0 +1,55 @@
+{{={= =}=}}
+import React, { useEffect } from 'react'
+import { useHistory } from 'react-router-dom'
+
+import config from '../../config.js'
+import api, { setAuthToken } from '../../api.js'
+
+// After a user authenticates via an Oauth 2.0 provider, this is the page that
+// the provider should redirect them to, while providing query string parameters
+// that contain information needed for the API server to authenticate with the provider.
+// This page forwards that information to the API server and in response get a JWT,
+// which it stores on the client, therefore completing the OAuth authentication process.
+export default function OAuthCodeExchange(props) {
+  const history = useHistory()
+  // NOTE: Different auth methods will have different Wasp API server validation paths.
+  // This helps us reuse one component for various methods (e.g., Google, Facebook, etc.).
+  const apiServerUrlHandlingOauthRedirect = constructOauthRedirectApiServerUrl(props.pathToApiServerRouteHandlingOauthRedirect)
```

Comment:
I'd slide this into the `useEffect` callback. It's used only in there and saves the reader from thinking about whether location or any of it should be reactive.

Reply:
Good thinking, will do!